client/components/theme-type-badge/index.tsx

```diff
@@ -19,56 +19,51 @@ import './style.scss';
 
 interface Props {
 	canGoToCheckout?: boolean;
-	forcePremium?: boolean;
+	isLockedStyleVariation?: boolean;
 	siteId: number | null;
 	siteSlug: string | null;
 	themeId: string;
-	tooltipHeader?: string;
-	tooltipMessage?: string;
 }
 
 const ThemeTypeBadge = ( {
 	canGoToCheckout,
-	forcePremium,
+	isLockedStyleVariation,
 	siteId,
 	siteSlug,
 	themeId,
-	tooltipHeader,
-	tooltipMessage,
 }: Props ) => {
 	const translate = useTranslate();
-	const _type = useSelector( ( state ) => getThemeType( state, themeId ) );
-	const type = forcePremium ? PREMIUM_THEME : _type;
+	const type = useSelector( ( state ) => getThemeType( state, themeId ) );
 
 	useEffect( () => {
-		if ( type === FREE_THEME ) {
+		if ( type === FREE_THEME && ! isLockedStyleVariation ) {
 			return;
 		}
 		recordTracksEvent( 'calypso_upgrade_nudge_impression', {
 			cta_name: 'theme-upsell',
 			theme: themeId,
 		} );
```

Comment:
I found this part a bit odd, I know it's not strictly part of this review, but wouldn't it be more simple to:
```suggestion
		if ( type !== FREE_THEME && isLockedStyleVariation ) {
			recordTracksEvent( 'calypso_upgrade_nudge_impression', {
				cta_name: 'theme-upsell',
				theme: themeId,
			} );
		}
```

Reply:
I always find harder to read code that is indented due to `if` statements, so (if possible) I have the preference of inverting the condition and return early to keep the indentation shallow.

See pbOQVh-29D-p2

Reply:
That's usually the case when more than one thing is happening. In this specific case, there is only one condition/outcome.

Reply:
Yeah, the problem is more obvious where are more than one check, but even with just one I still prefer an early return and avoid the extra indentation. I find it easier to read.